tests/Integration/PimApi/PimCatalogApiClientTest.php

```diff
@@ -0,0 +1,207 @@
+        );
+
+        $result = $this->pimCatalogApiClient->getCatalogs();
+
+        $this->assertEquals([
+            new Catalog(
+                '70313d30-8316-41c2-b298-8f9e7186fe9a',
+                'Catalog with product value filters',
+                true,
+            ),
+            new Catalog(
+                '8a8494d2-05cc-4b8f-942e-f5ea7591e89c',
+                'Catalog with attribute mapping',
+                false,
+            ),
+        ], $result);
+    }
+
+    /**
+     * @test
+     */
+    public function itReturnsAnEmptyListWhenNoCatalogsAreReturned(): void
+    {
+        $this->mockPimAPIResponse(
+            'get-catalogs-empty-list.json',
+            'https://example.com/api/rest/v1/catalogs',
+        );
+
+        $result = $this->pimCatalogApiClient->getCatalogs();
+
+        $this->assertEmpty($result);
+    }
+
+    /**
+     * @test
+     */
+    public function itThrowsWhenAnErrorOccurOnCatalogsRetrieval(): void
+    {
+        $this->mockHttpResponse(
+            'GET',
+            'https://example.com/api/rest/v1/catalogs',
+            [],
+            new MockResponse('', ['http_code' => 400]),
+        );
+
+        $this->expectException(PimApiException::class);
+        $this->pimCatalogApiClient->getCatalogs();
+    }
+
+    /**
+     * @test
+     */
+    public function itCreatesACatalogWithAName(): void
+    {
+        $this->mockHttpResponse(
+            'POST',
+            'https://example.com/api/rest/v1/catalogs',
+            [],
+            new MockResponse(\json_encode([
+                'id' => '7e018bfd-00e1-4642-951e-4d45684b51f4',
+                'name' => 'Demo App catalog',
+                'enabled' => false,
+            ], JSON_THROW_ON_ERROR),
+            ['http_code' => 201]
+            )
+        );
+
+        $result = $this->pimCatalogApiClient->createCatalog('Demo App catalog');
+
+        $this->assertEquals(new Catalog(
+            '7e018bfd-00e1-4642-951e-4d45684b51f4',
+            'Demo App catalog',
+            false,
+        ), $result);
+    }
+
+    /**
+     * @test
+     */
+    public function itThrowsWhenItFailsToCreateACatalog(): void
+    {
+        $this->mockHttpResponse(
+            'POST',
+            'https://example.com/api/rest/v1/catalogs',
+            [],
+            new MockResponse('', ['http_code' => 400]),
+        );
+
+        $this->expectException(PimApiException::class);
+        $this->pimCatalogApiClient->createCatalog('Demo App catalog');
+    }
+
+    /**
+     * @test
+     */
+    public function itSetsProductMappingSchemaForACatalog(): void
+    {
+        $catalogId = '7e018bfd-00e1-4642-951e-4d45684b51f4';
+        $this->mockHttpResponse(
+            'PUT',
+            "https://example.com/api/rest/v1/catalogs/$catalogId/mapping-schemas/product",
+            [],
+            new MockResponse('', ['http_code' => 204])
+        );
+
+        $this->expectNotToPerformAssertions();
```

Comment:
I didn't find an assertion like `assertNoExceptionAreThrown` that will be really great for this use case. But If you add at the end of the test an `$this->assertTrue(true);` I think if there is an exception thrown it will fail the test. Can you try it, please? If it works, I think it's a more robust solution (you can add a comment to explain why you are doing this assert).